src/os/data/layersyncdescriptor.js

```diff
@@ -142,17 +142,31 @@ os.data.LayerSyncDescriptor.prototype.getLayerOptions = goog.abstractMethod;
 os.data.LayerSyncDescriptor.prototype.getOptions = function() {
   var options = this.getLayerOptions();
   if (options) {
+    options['defaults'] = this.extractConfigDefaults(options);
```

Comment:
This line needs to be done after the `if` which follows it and in a form that works on arrays. This function is given either a single `options` or an array of them. This line as is does not work for the array case.